pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java

```diff
@@ -94,7 +95,15 @@ public enum AggregationFunctionType {
 
   // boolean aggregate functions
   BOOLAND("boolAnd"),
-  BOOLOR("boolOr");
+  BOOLOR("boolOr"),
+
+  // argMin and argMax
+  ARGMIN("argMin"),
+  ARGMAX("argMax"),
+  PINOTPARENTAGGREGATIONARGMIN(CommonConstants.RewriterConstants.PARENT_AGGREGATION_NAME_PREFIX + ARGMIN.getName()),
+  PINOTPARENTAGGREGATIONARGMAX(CommonConstants.RewriterConstants.PARENT_AGGREGATION_NAME_PREFIX + ARGMAX.getName()),
+  PINOTCHILDAGGREGATIONARGMIN(CommonConstants.RewriterConstants.CHILD_AGGREGATION_NAME_PREFIX + ARGMIN.getName()),
+  PINOTCHILDAGGREGATIONARGMAX(CommonConstants.RewriterConstants.CHILD_AGGREGATION_NAME_PREFIX + ARGMAX.getName());
```

Comment:
Can 103-106 be encapsulated within the rewriter itself as opposed to exposing them in AggregationFunctionType.java ? 

I feel we should only have user exposed aggregation functions which can be used in SQL in this file ?

Reply:
> I feel we should only have user exposed aggregation functions which can be used in SQL in this file ?
Do we have a specific reason for doing this?

Reply:
Main reason is clean interface. AggregationFunctionType is for user exposed in-built functions ideally.

So as a follow-up we should try to see how we can do this cleanly in future otherwise this file will end up having mix of things imo.